hudi-client/hudi-spark-client/src/main/scala/org/apache/spark/sql/HoodieInternalRowUtils.scala

```diff
@@ -202,7 +202,7 @@ object HoodieInternalRowUtils {
                                 renamedColumnsMap: JMap[String, String],
                                 fieldNameStack: JDeque[String]): RowFieldUpdater = {
     (newDataType, prevDataType) match {
-      case (newType, prevType) if prevType == newType =>
+      case (newType, prevType) if prevType.sql == newType.sql =>
```

Comment:
Could you add a few tests in this change, e.g., making nested types nullable?

Reply:
Sure

Reply:
Yeah, we need a test case.

Reply:
@yihua @danny0405 added a test. Confirmed it fails w/o the patch and passes  w/ it